lib/hammer_cli_foreman/ping.rb

```diff
@@ -19,9 +19,12 @@ def execute
         response = send_request
         print_data(response)
 
-        return 1 if HammerCLIForeman::CommandExtensions::Ping.failed?(response)
+        return HammerCLI::EX_OK unless HammerCLIForeman::CommandExtensions::Ping.failed?(response)
 
-        HammerCLI::EX_OK
```

Comment:
I like the exit if it's abnormal and closing with `EX_OK`. That implies `EX_OK` is the expected execution. Your version implies exiting with 1 is normal.

Reply:
Hmm... I thought that exit with `EX_OK` `unless` something failed should be done right away and don't check for which services will be shown and which won't.

I can refactor that, but I'm not sure if it will be more readable though..

Reply:
When I read a function (and this was also what previously happened) is that I'd expect all steps in between would error out early.

So roughly:
```ruby
return 1 if failure_condition_1
return 2 if failure_condition_2

# Everything went OK

0
```

This makes it easier to extend with more failure conditions. In this case, I'd suggest:
```ruby
if HammerCLIForeman::CommandExtensions::Ping.failed?(response)
  HammerCLIForeman::CommandExtensions::Ping.check_for_unrecognized(
     response, output_definition
  )
  return 1
end

HammerCLI::EX_OK
```

IMHO the flow is easier to follow: we see a failure, do some additional processing and return early. Does that make sense or am I explaining it poorly?

Reply:
@ekohl, I just was going to update the PR :D

Yes, that makes sense.